aspnet/dnx/projects.rst

```diff
@@ -1,8 +1,652 @@
++---------------------+-------------------------------------+-----------------------------------+
+| excludeBuiltIn      | ``bin/**;obj/**;*.kproj``           |                                   |
++---------------------+-------------------------------------+-----------------------------------+
+| compileFiles        |                                     | Wildcard is not allowed           |
++---------------------+-------------------------------------+-----------------------------------+
+| contentFiles        |                                     | Wildcard is not allowed           |
++---------------------+-------------------------------------+-----------------------------------+
+| preprocessFiles     |                                     | Wildcard is not allowed           |
++---------------------+-------------------------------------+-----------------------------------+
+| resourceFiles       |                                     | Wildecard is not allowed          |
++---------------------+-------------------------------------+-----------------------------------+
+| sharedFiles         |                                     | Wildecard is not allowed          |
++---------------------+-------------------------------------+-----------------------------------+
+
+Precedence
+^^^^^^^^^^
+
+The sequence of searching are:
+
+1. Gather files from include patterns
+2. Exclude files from ignore patterns
+3. Exclude files from includes of mutually exclusive types (see below)
+4. Adding individually specified files
+
+The following describes the exact lists that are built up with the following notation:
+
+* \+ means included
+* \- means excluded
+* glob() means the values are used in the globbing algorithm.
+
+::
+
+    CompileList =
+      +Glob( +compile +compileBuiltIn -compileExclude -exclude -excludeBuiltIn) 
+      -SharedList
+      -PreprocessList
+      +compileFiles
+    
+    PreprocessList =
+      +Glob( +preprocess -preprocessExclude -exclude -excludeBuiltIn) 
+      +preprocessFiles
+    
+    SharedList =
+      +Glob( +shared -sharedExclude -exclude -excludeBuiltIn) 
+      +sharedFiles
+    
+    ResourceList =
+      +Glob( +resource -resourceExclude -exclude -excludeBuiltIn) 
+      +resourceFiles
+    
+    ContentList =
+      +Glob( +content -contentExclude -exclude -excludeBuiltIn) 
+      -CompileList
+      -PreprocessList
+      -SharedList
+      -ResourcesList
+      +contentFiles
+    
+    BundleExcludeList =
+      +Glob ( +bundleExclude )
+
+Sharing files
+^^^^^^^^^^^^^
+
+The *shared* section of the *project.json* is designed to allow you to create a project that shares its source with other projects, rather than being compiled to a binary.
+
+The scenario where you would most commonly use this is when you have files such as ``sharedAssemblyInfo.cs`` where you want each project that depends on it to compile that file, rather than depend on a binary that exposes a type for you to consume.
```

Comment:
This isn't really for SharedAssemblyInfo. For that, projects would use the `compile` glob above.

This is designed for using a project to produce a package that contains source code, not binaries, such as in https://github.com/aspnet/dnx/tree/dev/src/Microsoft.Framework.CommandLineUtils.Sources